Repeated Group: Questionnaire Response validator fails if response contains multiple answers

**Describe the bug**
Suppose a questionnaire response json is passed as a argument with multiple answers for repeated group item. The Questionnaire response validator throws exception.
`    Caused by: java.lang.IllegalArgumentException: Missing questionnaire item for questionnaire response item question-1
E/AndroidRuntime:     at com.google.android.fhir.datacapture.validation.QuestionnaireResponseValidator.checkQuestionnaireResponseItems(QuestionnaireResponseValidator.kt:216)
        at com.google.android.fhir.datacapture.validation.QuestionnaireResponseValidator.checkQuestionnaireResponseItem(QuestionnaireResponseValidator.kt:236)
        at com.google.android.fhir.datacapture.validation.QuestionnaireResponseValidator.checkQuestionnaireResponseItems(QuestionnaireResponseValidator.kt:222)
        at com.google.android.fhir.datacapture.validation.QuestionnaireResponseValidator.checkQuestionnaireResponse(QuestionnaireResponseValidator.kt:202)
        at com.google.android.fhir.datacapture.QuestionnaireViewModel.<init>(QuestionnaireViewModel.kt:119)
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.java:120)`

**Component**
SDC library

**To Reproduce**
Steps to reproduce the behavior:
Use following questionnaire and questionnaire response
 [component_repeated_group.txt](https://github.com/google/android-fhir/files/10906403/component_repeated_group.txt)
[component_repeated_group_response_multiple.txt](https://github.com/google/android-fhir/files/10906402/component_repeated_group_response_multiple.txt)

@jingtang10

--- a/datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireResponses.kt
+++ b/datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireResponses.kt
@@ -67,16 +67,8 @@ private fun List<QuestionnaireResponse.QuestionnaireResponseItemComponent>.packR
 
       val questionnaireItem = questionnaireItems.single()
 
-      questionnaireResponseItems.forEach { it ->
-        if (questionnaireItem.type == Questionnaire.QuestionnaireItemType.GROUP) {
-          if (questionnaireItem.repeats) {
-            it.answer.forEach { it.item = it.item.packRepeatedGroups(questionnaireItem.item) }
-          } else {
-            it.item = it.item.packRepeatedGroups(questionnaireItem.item)
-          }
-        } else {
-          it.answer.forEach { it.item = it.item.packRepeatedGroups(questionnaireItem.item) }
-        }
+      questionnaireResponseItems.forEach {
+        it.item = it.item.packRepeatedGroups(questionnaireItem.item)
       }
 
       if (

--- a/datacapture/src/test/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireResponsesTest.kt
+++ b/datacapture/src/test/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireResponsesTest.kt
@@ -261,6 +261,121 @@ class MoreQuestionnaireResponsesTest {
   }
 
   @Test
+  fun `should pack repeated groups recursively`() {
+    val questionnaire =
+      Questionnaire().apply {
+        addItem(
+          QuestionnaireItemComponent().apply {
+            linkId = "repeated-group"
+            type = Questionnaire.QuestionnaireItemType.GROUP
+            repeats = true
+            addItem(
+              QuestionnaireItemComponent().apply {
+                linkId = "nested-repeated-group"
+                type = Questionnaire.QuestionnaireItemType.GROUP
+                repeats = true
+                addItem(
+                  QuestionnaireItemComponent().apply {
+                    linkId = "nested-nested-question"
+                    type = Questionnaire.QuestionnaireItemType.BOOLEAN
+                  },
+                )
+              },
+            )
+          },
+        )
+      }
+
+    val questionnaireResponse =
+      QuestionnaireResponse().apply {
+        addItem(
+          QuestionnaireResponseItemComponent().apply {
+            linkId = "repeated-group"
+            addItem(
+              QuestionnaireResponseItemComponent().apply {
+                linkId = "nested-repeated-group"
+                addItem(
+                  QuestionnaireResponseItemComponent().apply {
+                    linkId = "nested-nested-question"
+                    addAnswer(
+                      QuestionnaireResponseItemAnswerComponent().apply {
+                        value = BooleanType(true)
+                      },
+                    )
+                  },
+                )
+              },
+            )
+            addItem(
+              QuestionnaireResponseItemComponent().apply {
+                linkId = "nested-repeated-group"
+                addItem(
+                  QuestionnaireResponseItemComponent().apply {
+                    linkId = "nested-nested-question"
+                    addAnswer(
+                      QuestionnaireResponseItemAnswerComponent().apply {
+                        value = BooleanType(false)
+                      },
+                    )
+                  },
+                )
+              },
+            )
+          },
+        )
+      }
+
+    val packedQuestionnaireResponse =
+      QuestionnaireResponse().apply {
+        addItem(
+          QuestionnaireResponseItemComponent().apply {
+            linkId = "repeated-group"
+            addAnswer(
+              QuestionnaireResponseItemAnswerComponent().apply {
+                addItem(
+                  QuestionnaireResponseItemComponent().apply {
+                    linkId = "nested-repeated-group"
+                    addAnswer(
+                      QuestionnaireResponseItemAnswerComponent().apply {
+                        addItem(
+                          QuestionnaireResponseItemComponent().apply {
+                            linkId = "nested-nested-question"
+                            addAnswer(
+                              QuestionnaireResponseItemAnswerComponent().apply {
+                                value = BooleanType(true)
+                              },
+                            )
+                          },
+                        )
+                      },
+                    )
+                    addAnswer(
+                      QuestionnaireResponseItemAnswerComponent().apply {
+                        addItem(
+                          QuestionnaireResponseItemComponent().apply {
+                            linkId = "nested-nested-question"
+                            addAnswer(
+                              QuestionnaireResponseItemAnswerComponent().apply {
+                                value = BooleanType(false)
+                              },
+                            )
+                          },
+                        )
+                      },
+                    )
+                  },
+                )
+              },
+            )
+          },
+        )
+      }
+
+    questionnaireResponse.packRepeatedGroups(questionnaire)
+    assertResourceEquals(questionnaireResponse, packedQuestionnaireResponse)
+  }
+
+  @Test
   fun `should not modify non-repeated groups while packing repeated groups`() {
     val questionnaire =
       Questionnaire().apply {

